### ISSUE TITLE
Check if stomp reader correctly throttles publishers

### ISSUE THREAD

**Issue by hairyhum** [rabbitmq/rabbitmq-stomp#67]
Stomp reader works similar way to MQTT, so this issue https://github.com/rabbitmq/rabbitmq-mqtt/issues/65 can be here too.

**Comment by hairyhum** [comment]
STOMP reader use some kind of mix of MQTT and AMQP blocking mechanism.
`control_throttle` sill set connection state to `blocking` and it will be set to `blocked` only after `SEND` frame processing, and then in `blocked` state it's not calling `async_recv`.
So this means that two messages can be published before the reader becomes finally blocked.

**Comment by hairyhum** [comment]
@michaelklishin @dumbbell @essen any thoughts why it's waiting for `SEND` frame to be processed before blocking connection?

**Comment by michaelklishin** [comment]
@hairyhum that's more or less how AMQP 0-9-1 reader was supposed to do throttling a while ago. I don't see it as a significant problem: we only can know that a connection publishes after it does so once, and we can't really not process that publish since STOMP publishes are not split into multiple frames.

**Comment by hairyhum** [comment]
@michaelklishin so maybe we can allow one publish per connection by adding some flag. Currently there is one publish per blocking.

**Comment by michaelklishin** [comment]
@hairyhum it is "one publish per alarm" in AMQP 0-9-1. You cannot assume that a connection always publishes. The way throttling in STOMP works conceptually per your comment above seems fine to me. What I did observe an issue with in #68 is that connection doesn't seem to be blocked: I managed to publish 100s of messages from a STOMP gem REPL, and connections were not reported as blocked in the management UI. So there can be an implementation issue.
